diff -Naur pixman-0.28.2/Makefile.am pixman-0.28.2-patched/Makefile.am
--- pixman-0.28.2/Makefile.am	2012-12-02 13:37:49.000000000 -0500
+++ pixman-0.28.2-patched/Makefile.am	2013-02-13 13:56:41.000000000 -0500
@@ -1,4 +1,4 @@
-SUBDIRS = pixman demos test
+SUBDIRS = pixman
 
 pkgconfigdir=$(libdir)/pkgconfig
 pkgconfig_DATA=pixman-1.pc
diff -Naur pixman-0.28.2/Makefile.in pixman-0.28.2-patched/Makefile.in
--- pixman-0.28.2/Makefile.in	2012-12-10 06:47:46.000000000 -0500
+++ pixman-0.28.2-patched/Makefile.in	2013-02-13 13:53:44.000000000 -0500
@@ -273,7 +273,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = pixman demos test
+SUBDIRS = pixman
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = pixman-1.pc
 GPGKEY = 6FF7C1A8
